src/ring_theory/localization.lean

```diff
@@ -236,6 +236,47 @@ lemma exist_integer_multiples_of_finset (s : finset S) :
   ∃ (b : M), ∀ a ∈ s, is_integer R ((b : R) • a) :=
 exist_integer_multiples M s id
 
+/-- A choice of a common multiple of the denominators of a `finset`-indexed family of fractions. -/
+noncomputable
+def common_multiple {ι : Type*} (s : finset ι) (f : ι → S) : M :=
+(exist_integer_multiples M s f).some
+
+lemma common_multiple_mul_is_integer {ι : Type*} (s : finset ι) (f : ι → S)
+  (i : s) : is_integer R (common_multiple M s f • f i) :=
+(exist_integer_multiples M s f).some_spec i i.prop
```

Comment:
I was thinking to split this into a def + a lemma like for the finset-only case:
```suggestion
/-- The family of numerators after clearing the denominators of a finite set of fractions. -/
noncomputable
def integer_multiple {ι : Type*} [decidable_eq ι] (s : finset ι) (f : ι → S) (i : s) : R :=
((exist_integer_multiples M s f).some_spec i i.prop).some

lemma map_integer_multiple {ι : Type*} [decidable_eq ι] (s : finset ι) (f : ι → S) (i : s) :
  algebra_map R S (integer_multiple M s f i) =
    common_multiple M s f • f i :=
by { simp only [integer_multiple, dif_pos i.prop],
     exact ((exist_integer_multiples M s f).some_spec _ i.prop).some_spec }
```

Reply:
The code for the finset-only case would be:
```lean

/-- A choice of a common multiple of the denominators of a finite set of fractions. -/
noncomputable
def common_multiple_of_finset (s : finset S) : M :=
common_multiple M s id

/-- The finset of numerators after clearing the denominators of a finite set of fractions. -/
noncomputable
def finset_integer_multiple [decidable_eq R] [decidable_eq S] (s : finset S) : finset R :=
s.attach.image (λ t, integer_multiple M s id t)

open_locale pointwise

lemma finset_integer_multiple_image [decidable_eq R] [decidable_eq S] (s : finset S) :
  algebra_map R S '' (finset_integer_multiple M s) =
    common_multiple_of_finset M s • s :=
begin
  delta finset_integer_multiple common_multiple,
  rw finset.coe_image,
  ext,
  split,
  { rintro ⟨_, ⟨x, -, rfl⟩, rfl⟩,
    rw [map_integer_multiple, id],
    exact set.mem_image_of_mem _ x.prop },
  { rintro ⟨x, hx, rfl⟩,
    exact ⟨_, ⟨⟨x, hx⟩, s.mem_attach _, rfl⟩, map_integer_multiple M s id _⟩ }
end
```